packages/react-input/src/components/Input/useInputStyles.ts

```diff
@@ -40,46 +48,110 @@ const useRootStyles = makeStyles({
+      borderBottomLeftRadius: theme.borderRadiusMedium,
+      borderBottomRightRadius: theme.borderRadiusMedium,
+
+      // Flat 2px border:
+      // By default borderBottom will cause little "horns" on the ends. The clipPath trims them off.
+      // (This could be done without trimming using `background: linear-gradient(...)`, but using
+      // borderBottom makes it easier for people to override the color if needed.)
+      borderBottomWidth: '2px',
+      borderBottomStyle: 'solid',
+      borderBottomColor: theme.colorCompoundBrandStroke,
+      clipPath: 'inset(calc(100% - 2px) 0 0 0)',
+
+      // Animation for focus OUT
+      transform: 'scaleX(0)',
+      transition: `transform ${motionDurations.ultraFast} ${motionCurves.accelerateMid}`,
+    },
+    ':focus-within:after': {
+      // Animation for focus IN
+      transform: 'scaleX(1)',
+      transition: `transform ${motionDurations.normal} ${motionCurves.decelerateMid}`,
+    },
+    ':focus-within:active:after': {
+      // This is if the user clicks the field again while it's already focused
+      borderBottomColor: theme.colorCompoundBrandStrokePressed,
     },
   }),
   small: theme => ({
     minHeight: fieldHeights.small,
-    padding: `0 ${horizontalSpacing.sNudge}`,
+    ...shorthands.padding(`0 ${horizontalSpacing.sNudge}`),
     ...contentSizes.caption1(theme),
   }),
   medium: theme => ({
     minHeight: fieldHeights.medium,
-    padding: `0 ${horizontalSpacing.mNudge}`,
+    ...shorthands.padding(`0 ${horizontalSpacing.mNudge}`),
     ...contentSizes.body1(theme),
   }),
   large: theme => ({
     minHeight: fieldHeights.large,
-    padding: `0 ${horizontalSpacing.m}`,
+    ...shorthands.padding(`0 ${horizontalSpacing.m}`),
     ...contentSizes[400](theme),
-    gap: horizontalSpacing.sNudge,
+    ...shorthands.gap(horizontalSpacing.sNudge),
   }),
   inline: {
     display: 'inline-flex',
   },
   outline: theme => ({
     background: theme.colorNeutralBackground1,
-    border: `1px solid ${theme.colorNeutralStroke1}`,
+    ...shorthands.border('1px', 'solid', theme.colorNeutralStroke1),
     borderBottomColor: theme.colorNeutralStrokeAccessible,
+    ':hover': {
+      ...shorthands.borderColor(theme.colorNeutralStroke1Hover),
+      borderBottomColor: theme.colorNeutralStrokeAccessibleHover,
+    },
+    // DO NOT add a space between the selectors! It changes the behavior of make-styles.
+    ':active,:focus-within': {
+      ...shorthands.borderColor(theme.colorNeutralStroke1Pressed),
+      borderBottomColor: theme.colorNeutralStrokeAccessiblePressed,
+    },
   }),
   underline: theme => ({
     background: theme.colorTransparentBackground,
-    borderRadius: 0, // corners look strange if rounded
-    borderBottom: `1px solid ${theme.colorNeutralStrokeAccessible}`,
+    ...shorthands.borderRadius(0), // corners look strange if rounded
+    borderBottomWidth: '1px',
+    borderBottomStyle: 'solid',
+    borderBottomColor: theme.colorNeutralStrokeAccessible,
```

Comment:
```suggestion
    ...shorthands.borderBottom('1px', 'solid', theme.colorNeutralStrokeAccessible),
```

This should also work 😉